libnetwork/networkdb/networkdb.go

```diff
@@ -766,9 +767,9 @@ func (nDB *NetworkDB) createOrUpdateEntry(nid, tname, key string, entry interfac
 
 // deleteEntry this function handles the deletion of entries into the local tree store.
 // It is also used to keep in sync the entries number of the network (all tables are aggregated)
-func (nDB *NetworkDB) deleteEntry(nid, tname, key string) (bool, bool) {
-	_, okTable := nDB.indexes[byTable].Delete(fmt.Sprintf("/%s/%s/%s", tname, nid, key))
-	_, okNetwork := nDB.indexes[byNetwork].Delete(fmt.Sprintf("/%s/%s/%s", nid, tname, key))
+func (nDB *NetworkDB) deleteEntry(nid, tname, key string) (okTable bool, okNetwork bool) {
+	nDB.indexes[byTable], _, okTable = nDB.indexes[byTable].Delete([]byte(fmt.Sprintf("/%s/%s/%s", tname, nid, key)))
+	nDB.indexes[byNetwork], _, okNetwork = nDB.indexes[byNetwork].Delete([]byte(fmt.Sprintf("/%s/%s/%s", nid, tname, key)))
```

Comment:
See my other comment; perhaps if we drop the `key` for these, we can use `DeletePrefix`;

```suggestion
func (nDB *NetworkDB) deleteEntry(nid, tname string) (okTable bool, okNetwork bool) {
	nDB.indexes[byTable], okTable = nDB.indexes[byTable].DeletePrefix([]byte(fmt.Sprintf("/%s/%s", tname, nid)))
	nDB.indexes[byNetwork], okNetwork = nDB.indexes[byNetwork].DeletePrefix([]byte(fmt.Sprintf("/%s/%s", nid, tname)))
```

We may want to do the actual delete outside of the WalkPrefix function in that case though to remove duplicates first (although it looks like `DeletePrefix` will just ignore if it was already deleted).

Reply:
What happens when for some unknown reason (something something timeCompensation that I don't understand) not all entries in a prefix are to be deleted? Then you cannot use DeletePrefix, you have to redo a walk for each entry.